### PR TITLE
print out a friendly verification of the window start/stop times

### PR DESCRIPTION
**Pull request by mattdrees**
This prints out something like "window will begin in 6 hours, 4 minutes and 38 seconds and will end in 12 hours, 4 minutes and 38 seconds".
It will help prevent mistakes like messing up a.m./p.m. and missing daylight savings adjustments

### CODE REVIEW

**Comment by mattdrees** [comment]
@hlbraddock can you review a small change for me?

**Comment by hlbraddock** [comment]
Is this a mistake that has been made before?

File: maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java [hunk 1]
```diff
@@ -86,9 +91,54 @@ public Failure(String message, IOException exception)
...
+            "ended",
+            "will end");
+
+        log.info("window " + beginPhrase + " and " + endPhrase);
+    }
+
+    private String getRelativeTimePhrase(
+        DateTime now,
+        DateTime begin,
+        String pastVerb,
```
Comment (hlbraddock) [inline maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java]: Such the grammarian.

File: maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java [hunk 2]
```diff
@@ -86,9 +91,54 @@ public Failure(String message, IOException exception)
             }
             log.info("successfully updated window on " + server);
         }
+        if (failures.isEmpty())
+        {
+            printConfirmationOfWindowBoundaries(window);
```
Comment (hlbraddock) [inline maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java]: When is this logged? When you send client configuration to the maintenance server?

**Comment by mattdrees** [comment]
I made the am/pm mixup mistake yesterday. See the Sev1 hipchat room.

File: maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java (hunk 2 above)
Comment (mattdrees) [inline maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java]: yes. The whole output looks like
```
$ java -jar ../target/maintenance-filter-controller-1-SNAPSHOT.jar --configurationFile obieeMaintenanceWindowUpdate.yml
INFO  MaintenanceControlClient - successfully updated window on http://hart-a341.net.ccci.org:8180/ss/maintenanceControl/obiee
INFO  MaintenanceControlClient - successfully updated window on http://hart-a342.net.ccci.org:8180/ss/maintenanceControl/obiee
INFO  MaintenanceControlClient - successfully updated window on http://hart-a341.net.ccci.org:8680/wsapi/maintenanceControl/obiee
INFO  MaintenanceControlClient - successfully updated window on http://hart-a342.net.ccci.org:8680/wsapi/maintenanceControl/obiee
All 4 updates succeeded
```

Comment (mattdrees) [inline maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java]: Doh, wrong copy/paste. Fixing it now.

File: maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java [hunk 3]
```diff
@@ -86,9 +91,54 @@ public Failure(String message, IOException exception)
...
+            period = new Period(now, begin, type);
+            verb = futureVerb + " in %s";
+        }
+        else
+        {
+            period = new Period(begin, now, type);
+            verb = pastVerb + " %s ago";
+        }
+        PeriodFormatter periodFormatter = PeriodFormat.getDefault();
+        return String.format(verb, periodFormatter.print(period));
```
Comment (hlbraddock) [inline maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java]: What happened to good old string concatenation?

Comment (mattdrees) [inline maintenance-filter-controller/src/main/java/org/ccci/maintenance/MaintenanceControlClient.java]: the thing that makes that hard is that in one case you put the formatted period at the end of the string, and in the other you put it at the beginning.

I guess I could rearrange this to use just concatenation, but it'd be more cluttered.

**Comment by hlbraddock** [comment]
+1 This looks good.

**Comment by mattdrees** [comment]
Thanks Lee!

**Pull request merged**